### PR TITLE
Add hidden fields option to sanitizeEntity both per model and on global response

### LINKED ISSUE #4342: Hide / make private timestamps, _id, id and __v with MongoDB
**Describe the bug**
There is no way with the beta to make private some of the attributes through the model's JSON

**Steps to reproduce the behavior**
- Create a a new model with timestamps for example.
- Try to add it manually to the attributes in the model's JSON.
- Make it private.
- The API response will still contain the timestamps.
(Also valid with id, _id, and __v)

**Expected behavior**
We should be able to hide those attributes from the API response

**System**
- Node.js version: v11.3.0
- NPM version: 6.5.0
- Strapi version: v3.0.0-beta.17.2
- Database: Mongodb
- Operating system: MacOS/Debian

### PR DESCRIPTION
**Pull request by domenicoven**
#### Description of what you did:
Restarting from the [PR #4914](https://github.com/strapi/strapi/pull/4914) and extended it to have hidden fields both per model and globally.
Also added unit tests to sanitizeEntity, which did not have any.

Fixes #4342

### CODE REVIEW

**Comment by alexandrebodin** [comment]
Hi @domenicoven  thank you for this PR. We really appricate the help :) 

I'm not sure I like the idea of adding yet another configuration for this though. As we already have a private key for fields. I would rather like we extend it so we can specify properties that aren't attributes like _id, created_at and such on a model basis only for now. 

What do you think of  `options.privateAttributes` in the models so it matches the current naming ?

**Comment by domenicoven** [comment]
Hi @alexandrebodin:

> What do you think of `options.privateAttributes` in the models so it matches the current naming ?

yep, this makes sense, and actually I also prefer it. I'll change the PR to reflect your suggestion.

However, I think it makes sense to have this in the global response config as well, so that it gets applied to all models when using MongoDB, which is the topic of issue #4342, which I was having troubles with.
What do you think?

**Comment by alexandrebodin** [comment]
> Hi @alexandrebodin:
> 
> > What do you think of `options.privateAttributes` in the models so it matches the current naming ?
> 
> yep, this makes sense, and actually I also prefer it. I'll change the PR to reflect your suggestion.
> 
> However, I think it makes sense to have this in the global response config as well, so that it gets applied to all models when using MongoDB, which is the topic of issue #4342, which I was having troubles with.
> What do you think?

I understand your wish to add this config. I'm not sure where it should be put though... We refactored the config loading a lot (your PR has conflicts btw ;) You should update and read the new config doc ;)) Once you have check the new doc let me know wher eyou suggest we put the config

**Comment by domenicoven** [comment]
> > Hi @alexandrebodin:
> > > What do you think of `options.privateAttributes` in the models so it matches the current naming ?
> > 
> > 
> > yep, this makes sense, and actually I also prefer it. I'll change the PR to reflect your suggestion.
> > However, I think it makes sense to have this in the global response config as well, so that it gets applied to all models when using MongoDB, which is the topic of issue #4342, which I was having troubles with.
> > What do you think?
> 
> I understand your wish to add this config. I'm not sure where it should be put though... We refactored the config loading a lot (your PR has conflicts btw ;) You should update and read the new config doc ;)) Once you have check the new doc let me know where you suggest we put the config

Read everything and had a look at the code. Surely the refactoring since I opened this PR is huge :).
The only viable option I see as of now could be to add to /config/server.js a new object:
options.globalPrivateAttributes default: []

Otherwise we could skip this global setting, but then users will have to configure for each and every model with the same list of options.privateAttributes, which is not DRY...

**Comment by alexandrebodin** [comment]
here is my proposal:

We document this options :

`./config/api.js`
```js
module.exports = {
  responses: {
    privateFields: []
  }
}
```

and in models: 

`./model.settings.json`
```json
{
  "options": {
    "privateFields": []
   }
}
```

**Comment by domenicoven** [comment]
> here is my proposal:
> 
> We document this options :
> 
> `./config/api.js`
> 
> ```js
> module.exports = {
>   responses: {
>     privateFields: []
>   }
> }
> ```
> 
> and in models:
> 
> `./model.settings.json`
> 
> ```json
> {
>   "options": {
>     "privateFields": []
>    }
> }
> ```

This is just lovely!
I'll for this

**Comment by alexandrebodin** [comment]
Hey @domenicoven I think your PR as a small issue :) you have got a lot of changes coming from master I believe :)

**Comment by domenicoven** [comment]
> Hey @domenicoven I think your PR as a small issue :) you have got a lot of changes coming from master I believe :)

OMG, I did try to resolve the conflicts in the branch.
I'll rebase and leave the conflicts open then? 
Sorry for the confusion, I'm new to oss development, don't really know what is the right way: should I leave the conflicts in the PR next time? I manually updated the two conflicting files to the latest version and added my changes there, they should not be in conflict anymore.

File: docs/3.0.0-beta.x/concepts/configurations.md [hunk 1]
```diff
@@ -183,6 +183,25 @@ module.exports = ({ env }) => ({
...
 | `admin.build.backend`    | URL that the admin panel and plugins will request                                                                                                           | string        |             |
 
+## API
+
+**Path —** `./config/api.js`.
+
+```js
+module.exports = {
+  responses: {
+    privateFields: ['_v', 'id', 'created_at'],
```
Comment (alexandrebodin) [inline docs/3.0.0-beta.x/concepts/configurations.md:193]: can you rename this to privateAttribues to be consistent :)

Comment (domenicoven) [reply docs/3.0.0-beta.x/concepts/configurations.md:193]: that is actually a mistake, the code looks for privateAttributes. 
before pushing this change, should I rebase the changes coming, form master?

Comment (alexandrebodin) [reply docs/3.0.0-beta.x/concepts/configurations.md:193]: Your PR looks fine now, so not necessary ;)

**Comment by lauriejim** [comment]
@alexandrebodin can you ping me when the PR will be ready, to have a review on the documentation.
Thanks.

**Comment by domenicoven** [comment]
> Thanks for the work you've done ! I think we should also think about how we translate this into the graphql plugin

@alexandrebodin: I pushed a change to have this also in graphQL plugin, please have a look.

**Comment by alexandrebodin** [comment]
Thank you @domenicoven Going to test this this week !

**Comment by lauriejim** [comment]
Hello! If we go forward with this PR the documentation has to be updated on the v3.x instead of the v3.0.0-beta

**Comment by domenicoven** [comment]
> Hello! If we go forward with this PR the documentation has to be updated on the v3.x instead of the v3.0.0-beta

Hello @lauriejim , is it something I can do? When I forked the code to create this  PR, there was no v3.x folder for documentation.  I'd have to update the branch to the latest master version to have that folder and update the documentation accordingly. Can you please advise what are the recommended steps here?

**Comment by dalbitresb12** [comment]
Is there anything I can help with? Would really love to have this before launching a new site to production.

**Comment by alexandrebodin** [comment]
Hi @domenicoven We made quite a few changes to this code in the upcoming feature. (check branch features/rbac). Once merge into master you will need to upgrade your PR

**Comment by domenicoven** [comment]
> Hi @domenicoven We made quite a few changes to this code in the upcoming feature. (check branch features/rbac). Once merge into master you will need to upgrade your PR

At this point, I'd wait that feature to be merged before doing any change

**Pull request closed without merging**